homeassistant/components/abode/cover.py

```diff
@@ -37,10 +35,10 @@ def is_closed(self) -> bool:
         """Return true if cover is closed, else False."""
         return not self._device.is_open
 
-    def close_cover(self, **kwargs: Any) -> None:
```

Comment:
We keep `kwargs` always in there such that we can expand the method signature without having to update all existing implementations of a method.

Reply:
I think that there is a much more broader discussion about this practice (started the disussion [here](https://github.com/home-assistant/architecture/discussions/1072), but 5 cents about this:

I agree about the capabilities that having `**kwargs` have to be expanded with new parameters, but how does this answer:

1. How often this happens to be worth using this practice?
2. Why for CoverEntity this is the case, but on some other methods on other Entity classes this is not the case ([example1](https://github.com/home-assistant/core/blob/52bd3efad9f85154324f7a0906fa1257562b60f6/homeassistant/components/climate/__init__.py#L721-L749), [example2](https://github.com/home-assistant/core/blob/52bd3efad9f85154324f7a0906fa1257562b60f6/homeassistant/components/automation/__init__.py#L636-L641),[example3](https://github.com/home-assistant/core/blob/52bd3efad9f85154324f7a0906fa1257562b60f6/homeassistant/components/alarm_control_panel/__init__.py#L174-L228)? Should there be consistency across all HA entity classes?

With the current approach, a user when having to implement a child Entity class would have the go thought the following process:

1. Which are the parameters that I can get so I can customize my `close_cover` logic?
2. Until they realised that [this line](https://github.com/home-assistant/core/blob/52bd3efad9f85154324f7a0906fa1257562b60f6/homeassistant/components/cover/__init__.py#L175) defines the parameters that get passed, and in this case always `**kwargs` will always be empty (but developers would have already paid off the cognitive cost of understanding this)

The alternative, makes it explicit: implement the logic to `close_cover` where no parameters are being given. The need to extend the signature can be done one-off when needed (which happens very rarely) with a simple search and replace in VS Code (similar on how this PR was generated) at a very low cost. I think the benefits of making the interface more understandable to developers who implement logic is worth it (happy to discuss it more on the [architecture discussion](https://github.com/home-assistant/architecture/discussions/1072))

Reply:
Yes, but there are still solutions on how to handle that even for external custom integrations without having to rely on `**kwargs`. The solution relies on the assumption that we are going to add a new parameter and not remove them, and when extending new parameters, most likely would be optional (with a default value). If a new mandatory parameter is needed most likely would translate into a new service call or method. An elegant solution could be rely in method signature inspection to decide which parameters to pass.

An example could be:

```python
class ClimateEntity:

  async def async_set_temperature(
    self,
    temperature: float,
    ...
    new_argument: str | None = None, # this is added into HA Core
) -> None:
  ...


### External component outside of Home Assistant
class ClimateEntity:

  async def async_set_temperature(
    self,
    temperature: float,
    ...) -> None: # no new argument defined
  ...

# on the core library, on the method responsible for passing the 
# parameters from the service schema into the corresponding 
# method, we have the logic that would inspect the arguments and # pass only the ones defined by the method

async def _handle_entity_call(
 ...
):
  func_spec = inspect.getfullargspec(func)
  date = {k: v for k, v in data.items() if k in func_spec.args}
  func(**data)
```

The last part is a more representative view of the logic, but not exactly how it would look like as there are more logic on the method. But I think it can illustrate how things can be make reliable externally, while we improve the interfaces internally.